sled-diagnostics/src/queries.rs

```diff
@@ -25,7 +30,27 @@ const ZONEADM: &str = "/usr/sbin/zoneadm";
 pub const DEFAULT_TIMEOUT: Duration = Duration::from_secs(10);
 
 pub trait SledDiagnosticsCommandHttpOutput {
-    fn get_output(self) -> String;
+    fn get_output(self) -> SledDiagnosticsQueryOutput;
+}
+
+#[derive(Clone, Debug, JsonSchema, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum SledDiagnosticsQueryOutput {
+    Success {
+        /// The command and it's arguments.
+        command: String,
+        /// Any stdout/stderr produced by the command.
+        stdio: String,
```

Comment:
if we're now returning structured output, would it make sense to start separating stderr and stdout? or is it more useful to keep interleaving them?

Reply:
I personally think it's more useful to have them interleaved but I am open to changing it if others think it's better.
My reasoning is being able to see where an error occurred in a command's output vs just that an error occurred at some point.